### PR TITLE
test(detection): add adapter tests and ignore local data directory

### PR DESCRIPTION
**Pull request by danielsamo-se**
<details>
<summary>Before submitting</summary>

- [x] Self-reviewed the code
- [ ] Updated documentation, follow Google-style
- [ ] Added docs entry for autogeneration (if new functions/classes)
- [x] Added/updated tests
- [x] All tests pass locally

</details>

## Description

Adds unit tests for the Detections.from_* adapters (YOLOv5, Ultralytics, YOLO-NAS)
in supervision/detection/core.py.

The tests focus on behavior and edge cases and avoid external dependencies
by using lightweight fakes and monkeypatching.

Refs #2108

## Type of Change

- 🧪 Test update
- 🔧 Chore (ignoring local data directory)

## Motivation and Context

Detection adapters are a critical integration layer, but several branches
were previously untested. Adding focused unit tests improves confidence
during refactoring and increases coverage in the detection module.

## Changes Made

- Added unit tests for the Detections.from_* adapter methods
- Covered empty and non-empty prediction cases
- Ignored local data directory to prevent accidental commits

## Testing

- [x] I have tested this code locally
- [x] I have added unit tests covering the relevant adapter behavior
- [x] All new and existing tests pass

## Additional Notes

Tests are scoped to behavior and edge cases rather than implementation details,
following the project contribution guidelines.

### CODE REVIEW

File: test/detection/test_from_adapters.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+
+import supervision.detection.core as detection_core
+from supervision.detection.core import CLASS_NAME_DATA_FIELD, Detections
+
+
+class _FakeTensor:
```
Comment (Borda) [inline test/detection/test_from_adapters.py]: Lets move all these Fake-like to the helper module inside test/ folder

Comment (danielsamo-se) [reply test/detection/test_from_adapters.py]: Moved the adapter fakes to test/helpers.py and updated the tests to import them from there.
